File: perl/registrant_alert_classic.pl
#!/usr/bin/perl

use LWP::Protocol::https;         # From CPAN
use LWP::Simple;                  # From CPAN
use URI::Encode qw( uri_encode ); # From CPAN
use strict;                       # Good practice
use warnings;                     # Good practice

my $base_url = "https://www.whoisxmlapi.com/registrant-alert-api/search.php";
my $term1 = "whois";
my $exclude_term1 = "domain";
my $exclude_term2 = "news";
my $username = "Your registrant alert api username";
my $password = "Your registrant alert api password";

#######################
# Use a JSON resource #
#######################
print "JSON\n---\n".getData("json");

#######################
# Use an XML resource #
#######################
print "XML\n---\n".getData("xml");

#######################
# Getting the Data    #
#######################
sub getData {
    my $format = $_[0];
    my $url = "$base_url?username=$username&password=$password&term1=$term1"
            . "&exclude_term1=$exclude_term1&exclude_term2=$exclude_term2"
            . "&output_format=$format";
    $url = uri_encode($url);

    print "Get data by URL: $url\n";
    # 'get' is exported by LWP::Simple;
    my $object = get($url);

    die "Could not get $base_url!" unless defined $object;
    return $object
}

File: node/registrant_alert_classic.js
var http = require('https');
var querystring = require('querystring');

var username = 'Your registrant alert api username';
var password = 'Your registrant alert api password';
var term1 = 'test';
var exclude_term1 = 'movie';
var exclude_term2 = 'online';
var format = 'json';

var url = 'https://www.whoisxmlapi.com/registrant-alert-api/search.php';

var params = {
    term1: term1,
    username: username,
    password: password,
    output_format: format,
    exclude_term1: exclude_term1,
    exclude_term2: exclude_term2
};

url = url + '?' + querystring.stringify(params);

http.get(url, function(response) {
    var str = '';
    response.on('data', function(chunk) {
        str += chunk;
    });
    response.on('end', function() {
        console.log(JSON.parse(str));
    });
}).end();